src/sparseml/keras/models/registry.py

```diff
@@ -128,21 +130,27 @@ def create_zoo_model(
             pretrained if isinstance(pretrained, str) else attributes.default_desc
         )
 
-        return Zoo.load_model(
-            attributes.domain,
-            attributes.sub_domain,
-            attributes.architecture,
-            attributes.sub_architecture,
-            KERAS_FRAMEWORK,
-            attributes.repo_source,
-            attributes.default_dataset
+        model_dict = {
+            "domain": attributes.domain,
+            "sub_domain": attributes.sub_domain,
+            "architecture": attributes.architecture,
+            "sub_architecture": attributes.sub_architecture,
+            "framework": KERAS_FRAMEWORK,
+            "repo": attributes.repo_source,
+            "dataset": attributes.default_dataset
             if pretrained_dataset is None
             else pretrained_dataset,
-            None,
-            sparse_name,
-            sparse_category,
-            sparse_target,
-        )
+            "sparse_tag": f"{sparse_name}-{sparse_category}",
+        }
+        # clear cache
+        CACHE_DIR = os.path.expanduser(os.path.join("~", ".cache", "sparsezoo"))
+        [
+            shutil.rmtree(cached_model_dir)
+            for cached_model_dir in glob.glob(os.path.join(CACHE_DIR, "*"))
+            if os.path.isdir(cached_model_dir)
+        ]
+        stub = model_dict_to_stub(**model_dict)
+        return Model(stub)
```

Comment:
where is this function used? We are now returning a different `Model` class so it will be accessed differently. are we updating the call sites as well?

Reply:
I did not manage to thoroughly check those files. They were not covered by tests, plus I gave low priority to tf/keras files.